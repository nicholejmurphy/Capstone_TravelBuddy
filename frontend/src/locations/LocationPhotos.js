import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import ImageList from "@material-ui/core/ImageList";
import ImageListItem from "@material-ui/core/ImageListItem";

const useStyles = makeStyles(() => ({
  photos: {
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "space-around",
    overflow: "hidden",
    marginBottom: "20px",
  },
  imageList: {
    width: "100%",
    flexWrap: "nowrap",
  },
}));

/** Shows location photos
 * - Recieves list of photos from parent: LocationDetails
 */
function LocationPhotos({ photos }) {
  const classes = useStyles();

  return (
    <div className={classes.photos}>
      <ImageList rowHeight={250} className={classes.imageList} cols={3}>
        {photos.map((p) => (
          <ImageListItem key={p.id} cols={2}>
            <img key={p.id} src={p.images.original.url} alt={p.caption} />
          </ImageListItem>
        ))}
      </ImageList>
    </div>
  );
}

export default LocationPhotos;
