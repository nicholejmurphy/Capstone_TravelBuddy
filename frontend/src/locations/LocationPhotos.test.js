import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import LocationPhotos from "./LocationPhotos";
import { MemoryRouter } from "react-router";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <LocationPhotos photos={[]} />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LocationPhotos photos={[]} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
