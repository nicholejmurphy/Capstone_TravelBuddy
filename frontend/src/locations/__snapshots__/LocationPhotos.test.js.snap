// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-photos-3"
  >
    <ul
      class="MuiImageList-root makeStyles-imageList-4"
      style="margin: -2px;"
    />
  </div>
</DocumentFragment>
`;
